File: 백준/Silver/10828. 스택/스택.java
import java.util.*;
import java.io.*;

public class Main{

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        int n = Integer.parseInt(br.readLine());
        Stack<String> st = new Stack<>();
        while (n --> 0){
            String[] s = br.readLine().split(" ");
            if (s[0].equals("push")) st.push(s[1]);
            else if (s[0].equals("pop")){
                if (st.empty()) bw.write("-1\n");
                else bw.write(st.pop() +"\n");
            }
            else if (s[0].equals("size")) bw.write(st.size()+"\n");
            else if (s[0].equals("empty")) {
                if (st.empty()) bw.write("1\n");
                else bw.write("0\n");
            }

            else if (s[0].equals("top")) {
                if (st.empty()) bw.write("-1\n");
                else bw.write(st.peek() + "\n");
            }
        }

        bw.flush();
    }
}

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <bits/stdc++.h>

using namespace std;


int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    stack<int> st;
    int n;
    cin >> n;
    while (n-->0){
        string s;
        cin >> s;
        if (s == "push"){
            int tmp;
            cin >> tmp;
            st.push(tmp);
        }

        else if(s == "pop"){
            if (st.empty()) cout << -1 << '\n';
            else {
                cout << st.top() << '\n';
                st.pop();
            }
        }

        else if(s == "size") cout << st.size() << '\n';

        else if(s == "empty") {
            if (st.empty()) cout << 1 << '\n';
            else cout << 0 << '\n';
        }

        else{
            if (st.empty()) cout << -1 << '\n';
            else cout << st.top() << '\n';
        }
    }
}